api/v1alpha1/discoveredcluster_types.go

```diff
@@ -33,6 +33,7 @@ type DiscoveredClusterSpec struct {
 	APIURL            string       `json:"apiUrl" yaml:"apiUrl"`
 	CreationTimestamp *metav1.Time `json:"creationTimestamp,omitempty" yaml:"creationTimestamp,omitempty"`
 	ActivityTimestamp *metav1.Time `json:"activityTimestamp,omitempty" yaml:"activityTimestamp,omitempty"`
+	Type              string       `json:"type,omitempty" yaml:"type,omitempty"`
```

Comment:
The Type field is currently defined as optional. That means if there is no Type set it will be absent from the object json. If Type is set as required, then it will be set in the json as an empty string. Verify this is the desired behavior.

Reply:
Good catch-- it should be required